notebook/static/notebook/js/notificationarea.js

```diff
@@ -242,27 +242,42 @@ define([
             $kernel_ind_icon.attr('class','kernel_dead_icon').attr('title','Kernel Dead');
             knw.danger(short, undefined, showMsg);
         });
+        
+        var change_favicon = function (src) {
+            var link = document.createElement('link'),
+                oldLink = document.getElementById('favicon');
+            link.id = 'favicon';
+            link.type = 'image/x-icon';
+            link.rel = 'shortcut icon';
+            link.href = oldLink.href.replace();
+            if (oldLink) document.head.removeChild(oldLink);
+            document.head.appendChild(link);
+        };
 
         this.events.on('kernel_starting.Kernel kernel_created.Session', function () {
-            window.document.title='(Starting) '+window.document.title;
+            // window.document.title='(Starting) '+window.document.title;
             $kernel_ind_icon.attr('class','kernel_busy_icon').attr('title','Kernel Busy');
             knw.set_message("Kernel starting, please wait...");
+            change_favicon('/static/base/images/favicon-busy.ico');
         });
 
         this.events.on('kernel_ready.Kernel', function () {
             that.save_widget.update_document_title();
             $kernel_ind_icon.attr('class','kernel_idle_icon').attr('title','Kernel Idle');
             knw.info("Kernel ready", 500);
+            change_favicon('/static/base/images/favicon.ico');
         });
 
         this.events.on('kernel_idle.Kernel', function () {
             that.save_widget.update_document_title();
```

Comment:
I'm not sure I follow... It current changes the favicon to the busy icon on `'kernel_busy.Kernel'` and back to normal on `'kernel_idle.Kernel'` same for `'kernel_starting.Kernel kernel_created.Session'` and `'kernel_ready.Kernel'`. If we remove this from kernel idle, then the favicon will indicate that the kernel is busy even after it's idle/done.